File: ros-beginners/roscpp_simple_publisher_subscriber/ros/src/simple_publisher.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

int main(int argc, char **argv)
{
  ros::init(argc, argv, "roscpp_simple_publisher_node");
  ros::NodeHandle nh;
  ros::Publisher publisher = nh.advertise<std_msgs::String>("roscpp_pub_sub/event_out", 1);
  ros::Rate loop_rate(10);
  while(ros::ok())
  {
    std_msgs::String msg;
    msg.data = "Kiriman data dari node roscpp_simple_publisher_node";
    publisher.publish(msg);
    ros::spinOnce();
    loop_rate.sleep();
    
  }

  return 0;

}

File: ros-beginners/rospy_simple_publisher_subscriber/ros/script/simple_publisher.py
#!/usr/bin/env python
"""
This script is a simple python node which publishes
a std_msgs string.
"""
#-*- encoding: utf-8 -*-
__author__ = 'mohammad wasil'

import rospy
from std_msgs.msg import String

if __name__ == '__main__':
    '''
    Initilize node and string type publish topic at the loop rate of 10Hz.
    '''
    
    rospy.init_node("rospy_simple_publisher_node", anonymous=False)
    rate = rospy.Rate(10)
    pub = rospy.Publisher("rospy_pub_sub/event_out", String, queue_size=1)
    
    while not rospy.is_shutdown():
        string_msg = "Kiriman data dari rospy_simple_publisher_node"
        pub.publish(string_msg)
        rate.sleep()

